kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/KubernetesAttributesExtractor.java

```diff
@@ -53,9 +53,10 @@
   private static final String KIND_GROUP = "/(?<kind>[^/?]+)";
   private static final String NAME_GROUP = "(/(?<name>[^/?]+))?";
   private static final String NAMESPACE_GROUP = "(/namespaces/(?<namespace>[^/]+))?";
+  private static final String STATUS_MAYBE_GROUP = "(/status)?";
```

Comment:
Checking the provided link (https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#subresources)

I see there are 2 subresources `/status` and `/scale`.

Maybe we could change this to:
```
private static final String SUBRESOURCE_MAYBE_GROUP = "(/((status)|(scale)))?";
```
or even (although maybe this will probably break things):
```
private static final String SUBRESOURCE_MAYBE_GROUP = "(/(?<subresource>[^/]+))?";
```

WDYT??

Reply:
Good call. I should have thought to include scale while I was at it. I think perhaps we should hardcode the 2 options. If they add more options in the future we could make it more flexible. But for now it doesn't seem worth the extra risk in terms of matching things we shouldn't.

Reply:
Good call. I should have thought to include scale while I was at it. I think perhaps we should hardcode the 2 options. If they add more options in the future we could make it more flexible. But for now it doesn't seem worth the extra risk in terms of matching things we shouldn't.

I've updated appropriately.